### ISSUE TITLE
torch.nn.functional.sigmoid produces inconsistent results on different device for complex inputs

### ISSUE THREAD

**Issue by rookieLiu2018** [pytorch/pytorch#159870]
### 🐛 Describe the bug

The `torch.nn.functional.sigmoid function` produces inconsistent results on CPU and GPU for complex inputs with a real part of negative infinity (-inf).

```python
import torch
import numpy as np
input = torch.tensor(complex(-np.inf, 1))

gpu_output = torch.nn.functional.sigmoid(input.cuda())
print("GPU output:")
print(gpu_output)

cpu_output = torch.nn.functional.sigmoid(input.cpu())
print("CPU output:")
print(cpu_output)
```
output:
```
GPU output:
tensor(0.+0.j, device='cuda:0')
CPU output:
tensor(nan+nanj)
```

### Versions

```
PyTorch version: 2.5.1+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: 14.0.0-1ubuntu1.1
CMake version: version 3.31.6
Libc version: glibc-2.35
```

**Comment by krastogi-in** [comment]
I would like to work on this issue
